cat.plugin.ts

```diff
@@ -0,0 +1,85 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+
+class Breed {
+    public name: string = '';
+    public id: string = '';
+}
+
+export class CatPlugin extends Plugin {
+    public name: string = 'Cat Plugin';
+    public description: string = 'Generates pictures of cats.';
+    public usage: string = 'cat <breed (optional)>';
+    public permission: ChannelType = ChannelType.Public;
+    public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+    private _API_URL: string = 'https://api.thecatapi.com/v1/';
+    private _breeds: Breed[] = [];
+
+    constructor(public container: IContainer) {
+        super();
+        //creates list of breeds
+        this.container.httpService
+            .get(`${this._API_URL}breeds`)
+            .then((response: IHttpResponse) => {
+                const breeds = response.data;
+
+                this._breeds = breeds.map((breedData: { name: string, id: string }) => {
+                    return {
+                        name: breedData.name.toLowerCase(),
+                        id: breedData.id.toLowerCase(),
+                    }
+                })
+            })
+            .catch(console.log);
+    }
+
+    public async execute(message: IMessage, args?: string[]) {
+        if (args != undefined && (args[0] === "breeds" || args[0] === "breed")) {
```

Comment:
i would change to check if `args && args[0].includes('breed')`

Reply:
I can't do the args by itself, as the program feaks out if args is undefined, however I have changed the later part to include.

Reply:
Changed the plan for this, changed it to that, and I just decline ann undefined items/ empty lists prior to this line.